### PR TITLE
feat: add idle timeout field

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- regenerate protobuf TS types to include idleTimeout
- add idle timeout display/input on agent create/edit
- validate Go duration input and omit blank values

## Testing
- npm run lint
- npm run test
- npm run typecheck
- npm run build

#50

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- npm run lint (no errors)
- npm run test (passed: 35, failed: 0, skipped: 0)
- npm run typecheck (pass)
- npm run build (pass)

File: src/lib/duration.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const GO_DURATION_PATTERN =
+  /^[-+]?(?:(?:\d+(?:\.\d+)?|\.\d+)(?:ns|us|\u00b5s|ms|s|m|h))+$/;
+
+export const GO_DURATION_HELP_TEXT =
+  'Go duration format: e.g. 30s, 5m, 1h30m. Default: 5m.';
+
+export function isValidGoDuration(value: string): boolean {
+  return GO_DURATION_PATTERN.test(value);
+}
```
Comment (noa-lucent) [inline src/lib/duration.ts:9]: [minor] `isValidGoDuration` uses a non-trivial regex with several edge-case branches (fractional values, sign prefixes, multi-unit combos). The repo already has a test pattern for lib utilities (e.g. `labels.test.ts`). Please add a test file `src/__tests__/duration.test.ts` covering at least: valid inputs (`30s`, `5m`, `1h30m`, `1.5h`, `.5s`, `-5m`, `+5m`), invalid inputs (empty string, bare number `30`, unknown unit `1d`, plain text `abc`), and the blank-input boundary (which is handled by the callers, but the regex should return `false` for `""`).

**Review by noa-lucent — COMMENTED** [review]
Clean, well-structured PR that follows existing patterns consistently. The idle timeout field is properly wired through create, edit, and view flows with correct protobuf optional semantics.

One minor request: add unit tests for the new `duration.ts` validation utility, consistent with the repo's existing test coverage for lib utilities.

File: src/lib/duration.ts (hunk 1 above)
Comment (casey-brooks) [reply src/lib/duration.ts:9]: Added duration utility tests in src/__tests__/duration.test.ts covering the valid/invalid cases you listed.

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- npm run lint (no errors)
- npm run test (passed: 37, failed: 0, skipped: 0)
- npm run typecheck (pass)
- npm run build (pass)

**Review by noa-lucent — DISMISSED** [review]
Tests look good — covers the valid/invalid cases well and follows the repo's existing test conventions. All prior feedback addressed. Approving.

**Review by vitramir — DISMISSED** [review]

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- npm run lint (no errors)
- npm run test (passed: 37, failed: 0, skipped: 0)
- npm run typecheck (pass)
- npm run build (pass)

**Review by noa-lucent — DISMISSED** [review]
Re-reviewed after the third commit (chore: untrack generated protos). The new changes are clean and well-executed:

- `src/gen/` added to `.gitignore`, all generated files removed from tracking.
- `buf.lock` committed for reproducible codegen.
- CI updated with `npm run generate` before lint/test/build, plus a guard step that fails if generated code is accidentally committed.
- Dockerfile updated to generate types at image build time.

The idle timeout feature code (commits 1–2) remains correct as previously reviewed. All prior feedback addressed. Approving.

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- docker-buildx build --load -t console-app:test . (pass)

**Review by noa-lucent — DISMISSED** [review]
Correct fix. `node:20-slim` strips CA certificates, which breaks the TLS fetch to `buf.build` during `npm run generate`. Installing `ca-certificates` with proper cleanup in the build stage resolves this cleanly. No impact on the runtime image. Approving.

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- npm run lint (no errors)
- npm run test (passed: 37, failed: 0, skipped: 0)
- npm run typecheck (pass)
- npm run build (pass)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,6 +22,9 @@ jobs:
       - name: Install dependencies
         run: npm ci
 
+      - name: Generate API types
```
Comment (noa-lucent) [inline .github/workflows/ci.yml:25]: [nit] The removed guard (`git ls-files src/gen`) was a cheap safety net against someone accidentally force-adding generated files past `.gitignore`. The `.gitignore` entry is the primary protection and is sufficient, so this isn't blocking — but the guard cost only a few seconds and made the "no tracked generated code" policy explicit in CI. Consider keeping it unless there was a specific reason to drop it.

**Review by noa-lucent — APPROVED** [review]
The Dockerfile and feature code remain correct. Left a nit on the removed CI guard — not blocking.

**Review by vitramir — APPROVED** [review]
